// @realgrid/vue-realgrid/dist/vue-realgrid.umd.js
(function (global, factory) {
  typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports) :
  typeof define === 'function' && define.amd ? define(['exports'], factory) :
  (factory((global.VueRealgrid = {})));
}(this, (function (exports) { 'use strict';

  /* eslint-disable */

  var script = {
    name: 'RealGrid',
    props: {
      gridId: {
        type: String,
        required: true,
      },
      gridRoot: {
          type: String,
          required: true,
          default: '/'
      }
    },
    data: function data() {
      return {
        dataProvider: null,
        gridView: null,
      }
    },
    mounted: function mounted() {
      RealGridJS.setRootContext(this.gridRoot);

      this.dataProvider = new RealGridJS.LocalDataProvider();
      this.gridView = new RealGridJS.GridView(this.gridId);
      this.gridView.setDataSource(this.dataProvider);

      this.$emit('rendered', this.dataProvider, this.gridView);
    },
  };

  /* script */
              var __vue_script__ = script;
              
  /* template */
  var __vue_render__ = function() {
    var _vm = this;
    var _h = _vm.$createElement;
    var _c = _vm._self._c || _h;
    return _c("div", { staticClass: "grid", attrs: { id: _vm.gridId } })
  };
  var __vue_staticRenderFns__ = [];
  __vue_render__._withStripped = true;

    /* style */
    var __vue_inject_styles__ = function (inject) {
      if (!inject) { return }
      inject("data-v-0a47b14d_0", { source: "\n.grid {\n  width: 100%;\n  height: 200px;\n}\n", map: {"version":3,"sources":["/Users/onlydel/Codes/vue/vue-realgrid/src/vue-realgrid.vue"],"names":[],"mappings":";AAuCA;EACA,YAAA;EACA,cAAA;CACA","file":"vue-realgrid.vue","sourcesContent":["<script>\n/* eslint-disable */\n\nexport default {\n  name: 'RealGrid',\n  props: {\n    gridId: {\n      type: String,\n      required: true,\n    },\n    gridRoot: {\n        type: String,\n        required: true,\n        default: '/'\n    }\n  },\n  data() {\n    return {\n      dataProvider: null,\n      gridView: null,\n    }\n  },\n  mounted() {\n    RealGridJS.setRootContext(this.gridRoot);\n\n    this.dataProvider = new RealGridJS.LocalDataProvider();\n    this.gridView = new RealGridJS.GridView(this.gridId);\n    this.gridView.setDataSource(this.dataProvider);\n\n    this.$emit('rendered', this.dataProvider, this.gridView);\n  },\n}\n</script>\n\n<template>\n  <div :id=\"gridId\" class=\"grid\" ></div>\n</template>\n\n<style>\n.grid {\n  width: 100%;\n  height: 200px;\n}\n</style>"]}, media: undefined });

    };
    /* scoped */
    var __vue_scope_id__ = undefined;
    /* module identifier */
    var __vue_module_identifier__ = undefined;
    /* functional template */
    var __vue_is_functional_template__ = false;
    /* component normalizer */
    function __vue_normalize__(
      template, style, script$$1,
      scope, functional, moduleIdentifier,
      createInjector, createInjectorSSR
    ) {
      var component = (typeof script$$1 === 'function' ? script$$1.options : script$$1) || {};

      // For security concerns, we use only base name in production mode.
      component.__file = "/Users/onlydel/Codes/vue/vue-realgrid/src/vue-realgrid.vue";

      if (!component.render) {
        component.render = template.render;
        component.staticRenderFns = template.staticRenderFns;
        component._compiled = true;

        if (functional) { component.functional = true; }
      }

      component._scopeId = scope;

      {
        var hook;
        if (style) {
          hook = function(context) {
            style.call(this, createInjector(context));
          };
        }

        if (hook !== undefined) {
          if (component.functional) {
            // register for functional component in vue file
            var originalRender = component.render;
            component.render = function renderWithStyleInjection(h, context) {
              hook.call(context);
              return originalRender(h, context)
            };
          } else {
            // inject component registration as beforeCreate hook
            var existing = component.beforeCreate;
            component.beforeCreate = existing ? [].concat(existing, hook) : [hook];
          }
        }
      }

      return component
    }
    /* style inject */
    function __vue_create_injector__() {
      var head = document.head || document.getElementsByTagName('head')[0];
      var styles = __vue_create_injector__.styles || (__vue_create_injector__.styles = {});
      var isOldIE =
        typeof navigator !== 'undefined' &&
        /msie [6-9]\\b/.test(navigator.userAgent.toLowerCase());

      return function addStyle(id, css) {
        if (document.querySelector('style[data-vue-ssr-id~="' + id + '"]')) { return } // SSR styles are present.

        var group = isOldIE ? css.media || 'default' : id;
        var style = styles[group] || (styles[group] = { ids: [], parts: [], element: undefined });

        if (!style.ids.includes(id)) {
          var code = css.source;
          var index = style.ids.length;

          style.ids.push(id);

          if (isOldIE) {
            style.element = style.element || document.querySelector('style[data-group=' + group + ']');
          }

          if (!style.element) {
            var el = style.element = document.createElement('style');
            el.type = 'text/css';

            if (css.media) { el.setAttribute('media', css.media); }
            if (isOldIE) {
              el.setAttribute('data-group', group);
              el.setAttribute('data-next-index', '0');
            }

            head.appendChild(el);
          }

          if (isOldIE) {
            index = parseInt(style.element.getAttribute('data-next-index'));
            style.element.setAttribute('data-next-index', index + 1);
          }

          if (style.element.styleSheet) {
            style.parts.push(code);
            style.element.styleSheet.cssText = style.parts
              .filter(Boolean)
              .join('\n');
          } else {
            var textNode = document.createTextNode(code);
            var nodes = style.element.childNodes;
            if (nodes[index]) { style.element.removeChild(nodes[index]); }
            if (nodes.length) { style.element.insertBefore(textNode, nodes[index]); }
            else { style.element.appendChild(textNode); }
          }
        }
      }
    }
    /* style inject SSR */
    

    
    var component = __vue_normalize__(
      { render: __vue_render__, staticRenderFns: __vue_staticRenderFns__ },
      __vue_inject_styles__,
      __vue_script__,
      __vue_scope_id__,
      __vue_is_functional_template__,
      __vue_module_identifier__,
      __vue_create_injector__,
      undefined
    );

  // Import vue component

  // install function executed by Vue.use()
  function install(Vue) {
    if (install.installed) { return; }
    install.installed = true;
    Vue.component('RealGrid', component);
  }

  // Create module definition for Vue.use()
  var plugin = {
    install: install,
  };

  // To auto-install when vue is found
  /* global window global */
  var GlobalVue = null;
  if (typeof window !== 'undefined') {
    GlobalVue = window.Vue;
  } else if (typeof global !== 'undefined') {
    GlobalVue = global.Vue;
  }
  if (GlobalVue) {
    GlobalVue.use(plugin);
  }

  // It's possible to expose named exports when writing components that can
  // also be used as directives, etc. - eg. import { RollupDemoDirective } from 'rollup-demo';
  // export const RollupDemoDirective = component;

  exports.default = component;

  Object.defineProperty(exports, '__esModule', { value: true });

})));
